migrations/20160324165302_tag_associations.js

```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+exports.up = function (knex, Promise) {
+  return Promise.join(
+    knex.schema.createTable('posts_tags', function (table) {
+      table.increments()
+      table.bigInteger('post_id').references('id').inTable('post')
+      table.bigInteger('tag_id').references('id').inTable('tags')
```

Comment:
all of these join tables should have uniqueness constraints on their foreign key pairs